cmd/cat/cat.go

```diff
@@ -11,126 +10,65 @@ import (
-	filter             *filter.Filter
-	showWarcHeader     bool
-	showProtocolHeader bool
-	showPayload        bool
+type catConfig struct {
+	Offset             int64
+	RecordNum          int
+	RecordCount        int
+	FileName           string
+	Filter             *filter.Filter
+	ShowWarcHeader     bool
+	ShowProtocolHeader bool
+	ShowPayload        bool
 }
 
-func NewCommand() *cobra.Command {
-	var cmd = &cobra.Command{
-		Use:   "cat",
-		Short: "Concatenate and print warc files",
-		Long:  ``,
-		Example: `# Print all content from a WARC file
-warc cat file1.warc.gz
-
-# Pipe payload from record #4 into the image viewer feh
-warc cat -n4 -P file1.warc.gz | feh -`,
-		RunE: parseArgumentsAndCallCat,
-	}
-
-	cmd.Flags().Int64P(flag.Offset, "o", -1, flag.OffsetHelp)
-	cmd.Flags().IntP(flag.RecordNum, "n", -1, flag.RecordNumHelp)
-	cmd.Flags().IntP(flag.RecordCount, "c", 0, flag.RecordCountHelp+" Defaults to show all records except if -o or -n option is set, then default is one.")
-	cmd.Flags().BoolP(flag.ShowWarcHeader, "w", false, flag.ShowWarcHeaderHelp)
-	cmd.Flags().BoolP(flag.ShowProtocolHeader, "p", false, flag.ShowProtocolHeaderHelp)
-	cmd.Flags().BoolP(flag.ShowPayload, "P", false, flag.ShowPayloadHelp)
-	cmd.Flags().StringArray(flag.RecordId, []string{}, flag.RecordIdHelp)
-	cmd.Flags().StringSliceP(flag.RecordType, "t", []string{}, flag.RecordTypeHelp)
-	cmd.Flags().StringP(flag.ResponseCode, "S", "", flag.ResponseCodeHelp)
-	cmd.Flags().StringSliceP(flag.MimeType, "m", []string{}, flag.MimeTypeHelp)
-
-	return cmd
-}
-
-func parseArgumentsAndCallCat(cmd *cobra.Command, args []string) error {
-	config := &conf{}
-	if len(args) == 0 {
-		return errors.New("missing file name")
-	}
-	config.fileName = args[0]
-	config.offset = viper.GetInt64(flag.Offset)
-	config.recordCount = viper.GetInt(flag.RecordCount)
-	config.recordNum = viper.GetInt(flag.RecordNum)
-	config.showWarcHeader = viper.GetBool(flag.ShowWarcHeader)
-	config.showProtocolHeader = viper.GetBool(flag.ShowProtocolHeader)
-	config.showPayload = viper.GetBool(flag.ShowPayload)
-
-	if (config.offset >= 0 || config.recordNum >= 0) && config.recordCount == 0 {
-		config.recordCount = 1
-	}
-	if config.offset < 0 {
-		config.offset = 0
-	}
-
-	config.filter = filter.NewFromViper()
-
-	if !(config.showWarcHeader || config.showProtocolHeader || config.showPayload) {
-		config.showWarcHeader = true
-		config.showProtocolHeader = true
-		config.showPayload = true
-	}
-	return runE(config)
-}
-
-func runE(c *conf) error {
-	readFile(c, c.fileName)
-	return nil
-}
-
-func readFile(c *conf, fileName string) {
-	wf, err := gowarc.NewWarcFileReader(fileName, c.offset, gowarc.WithBufferTmpDir(viper.GetString(flag.TmpDir)))
-	defer func() { _ = wf.Close() }()
+func listRecords(config *catConfig, fileName string) ([]gowarc.WarcRecord, error) {
+	warcFileReader, err := gowarc.NewWarcFileReader(fileName, config.Offset, gowarc.WithBufferTmpDir(viper.GetString(flag.TmpDir)))
```

Comment:
Maybe add `tmpDir` to the config struct so we don't need to import viper in the `cat.go` file.

Reply:
I think this is out of scope of this PR, as it now purely does refactoring